src/rtnl.rs

```diff
@@ -153,6 +154,40 @@ impl Ifinfomsg {
             rtattrs,
         }
     }
+
+    /// Set the link with the given index up (equivalent to `ip link set dev DEV up`)
+    pub fn up(
+        ifi_family: RtAddrFamily,
+        ifi_type: Arphrd,
+        ifi_index: libc::c_int,
+        rtattrs: Rtattrs<Ifla, Vec<u8>>,
+    ) -> Self {
+        Ifinfomsg {
+            ifi_family,
+            ifi_type,
+            ifi_index,
+            ifi_flags: vec![Iff::Up],
+            ifi_change: Iff::Up.into(),
+            rtattrs,
+        }
+    }
+
+    /// Set the link with the given index down (equivalent to `ip link set dev DEV down`)
+    pub fn down(
+        ifi_family: RtAddrFamily,
+        ifi_type: Arphrd,
+        ifi_index: libc::c_int,
+        rtattrs: Rtattrs<Ifla, Vec<u8>>,
+    ) -> Self {
+        Ifinfomsg {
+            ifi_family,
+            ifi_type,
+            ifi_index,
+            ifi_flags: Vec::new(),
+            ifi_change: Iff::Up.into(),
```

Comment:
This looks like a copy paste error.

Reply:
Actually let me briefly check this one it might not be a copy and paste error. I remember being surprised here too.

Reply:
Yes this is meant to be like this you can also see this in @little-dudes comment show e.g. the golang version https://github.com/jbaublitz/neli/pull/61#issuecomment-557836682